tests/test_client_request.py

```diff
@@ -1230,3 +1230,46 @@ def test_loose_cookies_types(loop: Any) -> None:
 def test_gen_default_accept_encoding(has_brotli: Any, expected: Any) -> None:
     with mock.patch("aiohttp.client_reqrep.HAS_BROTLI", has_brotli):
         assert _gen_default_accept_encoding() == expected
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "trust_env", "expected_auth"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            True,
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            False,
+            None,
+        ),
+        ("", "example.com", True, None),
+        ("", "example.com", False, None),
+    ],
+)
+def test_basicauth_from_netrc(
+    tmp_path: pathlib.Path,
+    monkeypatch: pytest.MonkeyPatch,
+    make_request: Any,
+    netrc_contents: str,
+    hostname: str,
+    trust_env: bool,
+    expected_auth: Optional[helpers.BasicAuth],
+):
+
+    netrc_file_path = tmp_path / ".netrc"
+    with open(netrc_file_path, "w") as f:
+        f.write(netrc_contents)
+        f.flush()
```

Comment:
```suggestion
    netrc_file_path.write_text(netrc_contents)
```

Reply:
Also, the flush appears to be redundant in the original code, as you are closing the file when you exit the `with` block.